lib/new_relic/agent/transaction.rb

```diff
@@ -261,6 +261,30 @@ def initialize(category, options)
         end
       end
 
+      def parent_thread_id
+        Thread.current.nr_parent_thread_id if Thread.current.respond_to?(:nr_parent_thread_id)
+      end
+
+      def current_segment
+        # if this thread id has a current segment use that
+        current_thread_id = Thread.current.object_id
+        return current_segment_by_thread[current_thread_id] if current_segment_by_thread[current_thread_id]
+
+        # if this thread id does not have a current segment, this is where we'll need thread parent
+        return current_segment_by_thread[parent_thread_id] if current_segment_by_thread[parent_thread_id]
+
+        # if both of those fail, fall back to starting thread current segment
+        return current_segment_by_thread[@starting_thread_id]
```

Comment:
This is slightly reading as a `case` statement to me. Any idea if there's a performance impact for using that instead? (or if it really is applicable here?)

Reply:
hmmm I'm not sure how we could use a case here really. Since case statements are usually checking a single variable against multiple potential values, but in this situation we are checking multiple hash entries to see if they just exist before using them. I'm not sure how that would work, but i'd be open to refactoring this area. It's kind of an awkward method, so when i tried refactoring it before i didn't end up with anything good, so i just put it back. I'd be happy to pair on it if you'd like.

Reply:
I'm not sure how a case statement would exactly work here either, but I'm also happy to pair if you want to test it out later. I consider this a low priority compared to other things, so let's table it unless we have some extra time!